[FLINK-12847] [Connectors / Kinesis] update flink-connector-kinesis to use Apache 2.0 licensed code (#9494)
apache/flink: docs/dev/connectors/kinesis.zh.md

</dependency>
{% endhighlight %}

> <span class="label label-danger">Attention</span> Prior to Flink version 1.10.0 the `flink-connector-kinesis{{ site.scala_version_suffix }}` has a dependency on code licensed under the [Amazon Software License](https://aws.amazon.com/asl/) (ASL).

Reviewer: let's not shorting this to ASL; it just invites ambiguity.

> Linking to the flink-connector-kinesis for the prior versions will include ASL licensed code into your application.

Reviewer: ```suggestion
Linking to the prior versions of flink-connector-kinesis will include this code into your application.
```
